src/word.rs

```diff
@@ -80,9 +80,9 @@ enum UWordBoundsState {
     Numeric,
     Katakana,
     ExtendNumLet,
-    Regional,
+    Regional(/* half */ bool),
```

Comment:
Do you think it would be more readable to split these into separate enum variants, `{ Regional, RegionalHalf, Zwj, ZwjTainted }`?